src/app/projects/components/projects/projects.component.ts

```diff
@@ -272,4 +281,9 @@ export class ProjectsComponent implements OnInit, OnDestroy {
   setBadPasswordMsg(showBadPassMgs: boolean) {
     this.showBadPasswordMsg = showBadPassMgs;
   }
+
+  checkAndSetIfProjectNameExists(value: string) {
+    this.projectName = value;
+    this.projectsModals.sampleProjectName.projectNameExists = this.projectList.some(project => project.name == this.projectName.trim());
```

Comment:
this would call the trim function on every entry. I'm assuming that we need to store the untrimmed value in projectName so i'd go with:
```suggestion
    this.projectName = value;
    const checkName = value.trim();
    this.projectsModals.sampleProjectName.projectNameExists = this.projectList.some(project => project.name == checkName);
```